presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -595,197 +598,242 @@ void PrestoTask::updateExecutionInfoLocked(
-          prestoOp.getOutputCalls,
-          prestoOp.getOutputWall,
-          prestoOp.getOutputCpu);
-      CpuWallTiming finishAndBackgroundTiming;
-      finishAndBackgroundTiming.add(veloxOp.finishTiming);
-      finishAndBackgroundTiming.add(veloxOp.backgroundTiming);
-      setTiming(
-          finishAndBackgroundTiming,
-          prestoOp.finishCalls,
-          prestoOp.finishWall,
-          prestoOp.finishCpu);
-
-      prestoOp.blockedWall = protocol::Duration(
-          veloxOp.blockedWallNanos, protocol::TimeUnit::NANOSECONDS);
-
-      prestoOp.userMemoryReservation = protocol::DataSize(
-          veloxOp.memoryStats.userMemoryReservation, protocol::DataUnit::BYTE);
-      prestoOp.revocableMemoryReservation = protocol::DataSize(
-          veloxOp.memoryStats.revocableMemoryReservation,
-          protocol::DataUnit::BYTE);
-      prestoOp.systemMemoryReservation = protocol::DataSize(
-          veloxOp.memoryStats.systemMemoryReservation,
-          protocol::DataUnit::BYTE);
-      prestoOp.peakUserMemoryReservation = protocol::DataSize(
-          veloxOp.memoryStats.peakUserMemoryReservation,
-          protocol::DataUnit::BYTE);
-      prestoOp.peakSystemMemoryReservation = protocol::DataSize(
-          veloxOp.memoryStats.peakSystemMemoryReservation,
-          protocol::DataUnit::BYTE);
-      prestoOp.peakTotalMemoryReservation = protocol::DataSize(
-          veloxOp.memoryStats.peakTotalMemoryReservation,
-          protocol::DataUnit::BYTE);
-
-      prestoOp.spilledDataSize =
-          protocol::DataSize(veloxOp.spilledBytes, protocol::DataUnit::BYTE);
-
-      if (veloxOp.operatorType == "HashBuild") {
-        prestoOp.joinBuildKeyCount = veloxOp.inputPositions;
-        prestoOp.nullJoinBuildKeyCount = veloxOp.numNullKeys;
+      if (prestoPipeline.inputPipeline) {
+        prestoTaskStats.rawInputPositions += prestoPipeline.rawInputPositions;
+        prestoTaskStats.rawInputDataSizeInBytes +=
+            prestoPipeline.rawInputDataSizeInBytes;
+        prestoTaskStats.processedInputPositions +=
+            prestoPipeline.processedInputPositions;
+        prestoTaskStats.processedInputDataSizeInBytes +=
+            prestoPipeline.processedInputDataSizeInBytes;
       }
-      if (veloxOp.operatorType == "HashProbe") {
-        prestoOp.joinProbeKeyCount = veloxOp.inputPositions;
-        prestoOp.nullJoinProbeKeyCount = veloxOp.numNullKeys;
+      if (prestoPipeline.outputPipeline) {
+        prestoTaskStats.outputPositions += prestoPipeline.outputPositions;
+        prestoTaskStats.outputDataSizeInBytes +=
+            prestoPipeline.outputDataSizeInBytes;
       }
 
-      if (!veloxOp.dynamicFilterStats.empty()) {
-        prestoOp.dynamicFilterStats = toPrestoDynamicFilterStats(veloxOp);
+      for (const auto& driverStat : veloxPipeline.driverStats) {
+        for (const auto& [name, value] : driverStat.runtimeStats) {
+          addRuntimeMetric(taskRuntimeStats, name, value);
```

Comment:
Do we want to keep runtime metrics when `summarize`? It seems like this is what Java does: https://github.com/prestodb/presto/blob/5d63246c7dbd93cf0f6f444780bddbc05545bdda/presto-main/src/main/java/com/facebook/presto/operator/TaskStats.java#L599

Collection of runtime stats can be turned off with the flag @spershin suggested.